js/controller.js

```diff
@@ -1145,10 +1146,26 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
 
     //create plugin container elements
     createPluginElements: function() {
+      //if playerControlsOverAds is true then we need to override the setting
+      //for showing the control bar during ads.
+      if (this.state.playerParam && this.state.playerParam.playerControlsOverAds) {
+        if (this.state.config && this.state.config.adScreen) {
+          this.state.config.adScreen.showControlBar = true;
+        }
+      }
+
       var fullClass = (this.state.config.adScreen.showControlBar ? "" : " oo-full");
       $("#" + this.state.elementId + " .oo-player-skin").append("<div class='oo-player-skin-plugins"+fullClass+"'></div><div class='oo-player-skin-plugins-click-layer"+fullClass+"'></div>");
       this.state.pluginsElement = $("#" + this.state.elementId + " .oo-player-skin-plugins");
       this.state.pluginsClickElement = $("#" + this.state.elementId + " .oo-player-skin-plugins-click-layer");
+
+      //if playerControlsOverAds is true, then we need to set the size of the
+      //elements to be the full size of the player and not end where the control bar begins.
+      if (this.state.playerParam && this.state.playerParam.playerControlsOverAds) {
```

Comment:
Does this need to be reset at some point in case there's a new video or settings that don't require this?

Reply:
I was considering this a player setting and not a video setting. So it wouldn't change between videos

Reply:
Do we support changing the skin between videos?

Reply:
Oh ok, I don't think we support that, so no need to reset then.